### PR TITLE
Imihtishamarif/hge 728 collection minting journey

### PR DESCRIPTION
**Pull request by ihtisham1211**
[Includes newer components, changes to existing ones.

### CODE REVIEW

File: pages/create-collection.tsx [hunk 1]
```diff
@@ -1,7 +1,89 @@
...
+    royaltyWallets: [''],
+    spinPercent: 0,
+    nfts: [],
+  });
+
+  const [stepOneErrors, setStepOneErrors] = useState<StepOneErrors>({
+    tokenNameError: true,
+    creatorNameError: true,
+    displayNameError: true,
+    spinPercentError: false,
```
Comment (aseidma) [inline pages/create-collection.tsx:25]: "Spin" should be "split" everywhere (both in UI and code).
These conditions should not have to be fulfilled if split-royalties is disabled.
Currently, if you don't split royaltyfees in a proper way, you can't continue -- even if you don't want to split them at all, which should be valid.

File: pages/create-collection.tsx [hunk 2]
```diff
@@ -1,7 +1,89 @@
...
+import Steps from '../components/common/Steps';
+import AddNftStep from '../components/createCollection/AddNftStep';
+import CollectionInfoStep from '../components/createCollection/CollectionInfoStep';
+import VerifyCollectionMint from '../components/createCollection/VerifyCollectionMint';
+import { CollectionForm, StepOneErrors } from '../utils/Interfaces';
 
-const createCollection: NextPage = function () {
-  return <div>collection flow</div>;
+const CreateNft: NextPage = function () {
+  const [step, setStep] = useState(2);
```
Comment (aseidma) [inline pages/create-collection.tsx:10]: Flow should start at step 1, not 2.

File: components/createNft/RoyaltiesSection.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+}: RoyaltiesSectionProps) {
+  const { spinPercentError } = formDataErrors;
+  const addInput = () => {
+    const newWallet = '';
+    setFormData({
+      ...formData,
+      royaltyWallets: [...formData.royaltyWallets, newWallet],
+    });
+  };
+  const deleteInput = (wallet: string) => {
```
Comment (aseidma) [inline components/createNft/RoyaltiesSection.tsx:32]: This deletes all wallets.

File: components/common/FileInformation.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+}) {
+  const [showContent, setShowContent] = useState(false);
+  return (
+    <div className="flex items-center shadow-lg px-2 rounded-sm">
+      <p className="text-sm font-bold w-3 ">{`${number + 1}.`}</p>
+      <div className="flex items-center justify-between p-2 rounded-md min-w-full">
+        <p className="text-sm font-bold w-1/3  truncate">{fileName}</p>
+        <p className="text-sm font-bold text-center  ">{fileType}</p>
+        <div className="flex gap-x-2 items-center">
+          <Tooltip
```
Comment (aseidma) [inline components/common/FileInformation.tsx:31]: Move delete icon to the far right of the component (behind the file size display).

File: components/createCollection/AddNftStep.tsx [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+              editNft={() => editNftItem(nft)}
+              image={URL.createObjectURL(nft.nftThumbnail!)}
+              nftName={nft.tokenName}
+              collectionName={nft.creatorName}
+            />
+          ))}
+        </div>
+        <Button
+          onClick={() => {
+            setOpenConfirmMintModal(true);
```
Comment (aseidma) [inline components/createCollection/AddNftStep.tsx:156]: Move confirm modal to "Verify and Mint" Screen. It should open after clicking "Mint now".

File: components/createNft/SellingOptionsSection.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            <p>{option.name}</p>
+          </div>
+        ))}
+      </div>
+      {formDataErrors.sellingOptionError && (
+        <ErrorMessage errorText="Please select one option" />
+      )}
+
+      {/* <HbarLogo /> */}
+      {currentOption?.name === 'Mint and Sell' && (
```
Comment (aseidma) [inline components/createNft/SellingOptionsSection.tsx:67]: Disable the "Mint and Sell" option for now, this is a feature that will be added in the near future.

File: components/createNft/RoyaltiesSection.tsx [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+        step={0.5}
+        defaultValue={formData?.spinPercent}
+        onChange={handleFormChange}
+        name="spinPercent"
+      />
+      <span className="flex items-center justify-between text-sm font-bold my-6">
+        <h5>Spin Royalties</h5>
+        <Switch
+          labelFor="spinRoyaltiesEnabled"
+          checked={formData.spinRoyaltiesEnabled}
```
Comment (aseidma) [inline components/createNft/RoyaltiesSection.tsx:94]: Make this false by default.

**Review by aseidma — CHANGES_REQUESTED** [review]
Thank you for the PR! Please see the requested changes additionally to Figma changes by @danielHype (please highlight which ones are new so Muhammad knows what to change).

Additionally, all NFTs are gone if you press the return button in the browser at any point in time. Could you please implement logic that saves the current state to local storage and keeps track of it there @ihtisham1211?

File: pages/create-collection.tsx (hunk 2 above)
Comment (ihtisham1211) [reply pages/create-collection.tsx:10]: done

File: components/common/FileInformation.tsx (hunk 4 above)
Comment (ihtisham1211) [reply components/common/FileInformation.tsx:31]: done

File: components/createNft/RoyaltiesSection.tsx (hunk 7 above)
Comment (ihtisham1211) [reply components/createNft/RoyaltiesSection.tsx:94]: Done

File: components/createNft/SellingOptionsSection.tsx (hunk 6 above)
Comment (ihtisham1211) [reply components/createNft/SellingOptionsSection.tsx:67]: done

File: components/common/Modal.tsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+}) {
+  const closeModal = () => {
+    setShowModal(false);
+    onClose();
+  };
+
+  return (
+    <div>
+      {showModal && (
+        <div className="inset-0 absolute flex-col items-center bg-gray-900/50  z-10 min-w-full px-[5%] md:px-[15%] pt-10 md:pt-40 min-h-[300vh]">
```
[the remark this answers is not in the document]
Comment (ihtisham1211) [reply components/common/Modal.tsx:25]: Done

File: components/common/FileInformation.tsx (hunk 4 above)
Comment (ihtisham1211) [reply components/common/FileInformation.tsx:31]: done

File: components/createCollection/AddNftStep.tsx (hunk 5 above)
Comment (ihtisham1211) [reply components/createCollection/AddNftStep.tsx:156]: done

File: components/createNft/RoyaltiesSection.tsx (hunk 3 above)
Comment (ihtisham1211) [reply components/createNft/RoyaltiesSection.tsx:32]: done

File: components/createNft/SellingOptionsSection.tsx (hunk 6 above)
Comment (ihtisham1211) [reply components/createNft/SellingOptionsSection.tsx:67]: done

File: pages/create-collection.tsx (hunk 1 above)
Comment (ihtisham1211) [reply pages/create-collection.tsx:25]: done

**Pull request merged**